### ISSUE TITLE
"pyblosxom-cmd test" doesn't return >0 on importErrors

### ISSUE THREAD

**Issue by Dieterbe** [pyblosxom/pyblosxom#8]
TLDR: read the last line of this post.
Hi,
I have a plugin like this (last function is the important bit):

```
# vim: tabstop=4 shiftwidth=4 expandtab
"""
Preformatter which uses pygments to highlight source code syntax
In your entries, you can highlight code by wrapping it in 'code' tags.
You can specify a language by using making the 'lang' attribute a lexer
shortname. See http://pygments.org/docs/lexers/
Otherwise, the most appropriate lexer will be guessed.

Examples:
* Within CDATA tags, you can literally put the source code (recommended)

<code><![CDATA[
your_source_code(<somevariable>); /* here */
echo ("<html><tag foo='bar'>");
]]></code>

* Otherwise, you need to replace html entities by their references
  (use this when you have (short) source code, without html entities)

<code lang="html">&lt;html&gt;</code>


Depends on pygments, lxml, beautiful-soup
"""
__author__ = 'Dieter Plaetinck <dieter@plaetinck.be>'
__version__ = "0.1"
__url__ = "http://pyblosxom.sourceforge.net/"
__description__ = "Preformatter which uses pygments to highlight source code syntax."
PREFORMATTER_ID = 'syntaxhighlight'

from pygments import highlight
from pygments.lexers import PythonLexer, get_lexer_by_name, guess_lexer
from pygments.formatters import HtmlFormatter
from lxml import etree
from lxml.html.soupparser import fromstring

def cb_start(args):
    request = args["request"]

def cb_preformat(args):
    """
    Preformat callback chain looks for this.

    @param args: a dict with 'parser' string and a list 'story'
    @type args: dict
    """
    if args['parser'] == PREFORMATTER_ID:
        return parse(''.join(args['story']))

def parse(text):
    """
    Find code and highlight it

    @param text: A text for conversion
    @type text: string
    """
    tree = fromstring(text)
    code = tree.xpath('//code')
    for el in code:
        highlightcallback (el)

    result = etree.tostring(tree)
    return result

def highlightcallback (code):
    try:
        lexer = get_lexer_by_name(code.attrib['lang'])
    except Exception, e:
        lexer = guess_lexer(etree.tostring(code))
    output = code.text_content() # same as `etree.tostring(code, method='text')` afaict
    output = highlight(output, lexer, HtmlFormatter())
    # NOTE: emitting the styles like this doesn't feel right
    # if you have multiple entries with source code -> redundant style tags
    # plus, all this style info doesn't really belong in the html
    output = '<style>' + HtmlFormatter().get_style_defs('.highlight') + '</style>' + output
    newElement = fromstring(output)
    code.clear()
    code.append(newElement)

def verify_installation(request):
    try:
        import lxml
        import pygments
        import lxml.html.soupparser
    except Exception, e:
        print "Missing dependencies: %s" % str(e)
        return 0
    return 1
```

The idea being, if you run `pyblosxom-cmd test`, it will exit >0 and print the error about missing dependencies.
However, here's the output when a dependency is missing:

```
$ pyblosxom-cmd test                                         
pyblosxom-cmd version 1.5-dev git-master
Trying to import the config module....
System Information
==================

- pyblosxom:    1.5-dev git-master
- sys.version:  2.7.2 (default, Jan 31 2012, 13:26:35)  [GCC 4.6.2
  20120120 (prerelease)]
- os.name:      posix
- codebase:     /usr/lib/python2.7/site-packages

Checking config.py file
=======================
- properties set: 31
- datadir '/srv/http/dieterblog/entries' exists.
- flavourdir '/srv/http/dieterblog/flavours' exists.

Checking plugin configuration
=============================
- Some plugins failed to load.

----
plugin:  syntaxhighlight
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/Pyblosxom/plugin_utils.py", line 105, in initialize_plugins
    _module = __import__(mem)
  File "/srv/http/dieterblog/plugins/syntaxhighlight.py", line 31, in <module>
    from pygments import highlight
ImportError: No module named pygments

----
- This goes through your plugins and asks each of them to verify
  configuration and installation.

----
plugin:  Pyblosxom.plugins.comments
file:    /usr/lib/python2.7/site-packages/Pyblosxom/plugins/comments.pyc
version: $Id$
missing optional property: 'comment_dir'
missing optional property: 'comment_ext'
missing optional property: 'comment_draft_ext'
missing optional property: 'comment_nofollow'
missing optional property: 'comment_disable_after_x_days'
PASS
----
plugin:  pymetatime
file:    /srv/http/dieterblog/plugins/pymetatime.pyc
version: 0.1 - 8/31/10
missing optional config property 'display_date' 
Allows you to specifiy the display date format
missing optional config property 'display_time' 
Allows you to specifiy the display time format
PASS
----
plugin:  Pyblosxom.plugins.magicword
file:    /usr/lib/python2.7/site-packages/Pyblosxom/plugins/magicword.pyc
version: 2011-10-28
PASS
----
plugin:  nopose
file:    /srv/http/dieterblog/plugins/nopose.pyc
version: 0.1
PASS
----
plugin:  Pyblosxom.plugins.pages
file:    /usr/lib/python2.7/site-packages/Pyblosxom/plugins/pages.pyc
version: 2011-10-22
PASS
----

The following plugins do not support installation verification:
- 'Pyblosxom.plugins.tags' (/usr/lib/python2.7/site-packages/Pyblosxom/plugins/tags.pyc)
- 'guid' (/srv/http/dieterblog/plugins/guid.pyc)
- 'readmore' (/srv/http/dieterblog/plugins/readmore.pyc)

Verification complete.  Correct any errors and warnings above.
```

so what really happens is, pyblosxom tries to load the plugin, but the same imports are at the top, which fail, so it doesn't load the plugin, it does give a warning about this, but still `exits(0)` at the end.
A solution could be to do all imports only in the functions that need them, to make the plugin get loaded and `verify_installation()` would probably work properly, but I think many people who write plugins could make the same mistake and put their imports in the beginning.  Maybe the simplest solution is just to `exit(1)` when configured plugins fail to load.
